torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -768,19 +759,69 @@ def _get_ignored_modules(
         )
         return ignored_modules
 
-    def _get_ignored_params(
-        self,
-        ignored_modules: Set[torch.nn.Module],
-    ) -> Set[torch.nn.Parameter]:
-        """
-        Returns the parameters of the modules in ``ignored_modules`` as a
+    def _get_ignored_parameters(self) -> Set[torch.nn.Parameter]:
+        """Returns the parameters of the modules in ``ignored_modules`` as a
         :class:`set`, excluding any :class:`FlatParameter` s.
         """
+        assert hasattr(self, "_ignored_modules"), \
+            "Expects `self._ignored_modules` to be initialized"
         return set(
-            p for m in ignored_modules for p in m.parameters()
+            p for m in self._ignored_modules for p in m.parameters()
             if not isinstance(p, FlatParameter)
         )
 
+    def _get_ignored_named_tensors(
+        self,
+        ignored_modules: Set[torch.nn.Module],
+        named_tensor_fn: Callable,
+    ) -> Set[Tuple[str, torch.Tensor]]:
+        """
+        This performs a module walk to get the full parameter and buffer names
+        depending on ``named_tensor_fn``, which should either be
+        ``named_parameters()`` or ``named_buffers()`. We require a separate
+        :meth:`_get_ignored_parameters` that does not use this module walk
+        since that method needs to be called in the FSDP constructor before any
+        wrapping occurs, which means that we cannot start a module walk from
+        ``self`` as in this method.
+        """
+        ignored_named_tensors = set()
+
+        def f(ignored_named_tensors, module: torch.nn.Module, prefix: str):
```

Comment:
Can we dedupe this with `def f()` in get_param_to_unflat_param_name? Seems like a fair amount of the code is duplicated but there are some slight differences.